app/helpers/application_filter_helper.rb

```diff
@@ -0,0 +1,248 @@
+module ApplicationFilterHelper
```

Comment:
Not a blocker for merging but wonder if this should be in `models` as well ... I guess technically this is more like a query service? Either way `app/helpers' may not be the best home for this ...